src/openrct2/network/Network.cpp

```diff
@@ -2532,40 +2539,64 @@ void Network::Server_Handle_TOKEN(NetworkConnection& connection, [[maybe_unused]
 void Network::Client_Handle_OBJECTS_LIST(NetworkConnection& connection, NetworkPacket& packet)
 {
     auto& repo = GetContext()->GetObjectRepository();
-    uint32_t size;
-    packet >> size;
-    log_verbose("client received object list, it has %u entries", size);
-    if (size > OBJECT_ENTRY_COUNT)
+
+    uint32_t index = 0;
+    uint32_t totalObjects = 0;
+    packet >> index >> totalObjects;
+
+    if (index == 0)
+    {
+        // Start of objects list.
+        _missingObjects.clear();
+    }
+
+    if (totalObjects > OBJECT_ENTRY_COUNT)
     {
         connection.SetLastDisconnectReason(STR_MULTIPLAYER_SERVER_INVALID_REQUEST);
         connection.Socket->Disconnect();
         log_warning("Server sent invalid amount of objects");
         return;
     }
-    std::vector<std::string> missingObjects;
-    for (uint32_t i = 0; i < size; i++)
+
+    char objectListMsg[256];
+    uint32_t args[] = {
+        index + 1,
+        totalObjects,
+    };
+    format_string(objectListMsg, 256, STR_MULTIPLAYER_RECEIVING_OBJECTS_LIST, &args);
+
+    auto intent = Intent(WC_NETWORK_STATUS);
+    intent.putExtra(INTENT_EXTRA_MESSAGE, std::string{ objectListMsg });
+    intent.putExtra(INTENT_EXTRA_CALLBACK, []() -> void { gNetwork.Close(); });
+    context_open_intent(&intent);
+
+    const char* name = reinterpret_cast<const char*>(packet.Read(8));
+    // Required, as packet has no null terminators.
+    std::string s(name, name + 8);
+    uint32_t checksum, flags;
```

Comment:
```suggestion
    uint32_t checksum = 0;
    uint32_t flags = 0;
```

To be consistent with something similar at the top of the function